govcd/edgegateway.go

```diff
@@ -200,82 +200,84 @@ func (eGW *EdgeGateway) RemoveNATPortRule(networkHref, natType, externalIP, exte
 
 }
 
-// Temporary fix to support parsing network Id from HREF
-// TODO: remove this function when ruleId is available
-func extractObjectIDfromPath(locationPath string) (string, error) {
-	if locationPath == "" {
-		return "", fmt.Errorf("unable to get ID from empty path")
+// RemoveNATRule removes NAT removes NAT rule identified by ID and handles task. Returns error if issues rise.
+// Old functions RemoveNATPortMapping and RemoveNATMapping removed using rule details
+// and expected interface to be of external network type.
+func (eGW *EdgeGateway) RemoveNATRule(id string) error {
+	task, err := eGW.RemoveNATRuleAsync(id)
+	if err != nil {
+		return fmt.Errorf("error removing DNAT rule: %#v", err)
 	}
-
-	cleanPath := path.Clean(locationPath) // Removes trailing slash if there is one
-	splitPath := strings.Split(cleanPath, "/")
-
-	if len(splitPath) < 2 {
-		return "", fmt.Errorf("path does not contain url path: %s", splitPath)
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
 	}
 
-	objectID := splitPath[len(splitPath)-1]
-
-	return objectID, nil
-}
-
-func (eGW *EdgeGateway) RemoveNATMapping(natType, externalIP, internalIP, port string) (Task, error) {
-	return eGW.RemoveNATPortMapping(natType, externalIP, port, internalIP, port)
+	return nil
 }
 
-func (eGW *EdgeGateway) RemoveNATPortMapping(natType, externalIP, externalPort, internalIP, internalPort string) (Task, error) {
-	// Find uplink interface
-	var uplink types.Reference
-	for _, gi := range eGW.EdgeGateway.Configuration.GatewayInterfaces.GatewayInterface {
-		if gi.InterfaceType != "uplink" {
-			continue
-		}
-		uplink = *gi.Network
+// RemoveNATRuleAsync removes NAT rule and return  or error.
```

Comment:
"and return or error" -> "or returns an error"?

Reply:
Fixed